### ISSUE TITLE
Error in Iso Area algorithm(s) when Project CRS != Layer(s) CRS

### ISSUE THREAD

**Issue by ghtmtt** [root676/QNEAT3#29]
Hi,

I've 2 layers (road network and starting points) both in EPSG:6707 (projected CRS). All the analysis of QNeat work perfectly. Then I loaded the OpenStreetMap XYZ Tile and changed the project CRS to EPSG:3857. 

QNeat algorithms stopped working. E.g. the following traceback is from ``Iso-Area as Contours (from Layer)``. Knowing that it is always better not to mix CRS I just wanted to report this behavior ;)

p.s. AWESOME plugin!

```
Traceback (most recent call last):
File "/home/matteo/.local/share/QGIS/QGIS3/profiles/matteo/python/plugins/QNEAT3/algs/IsoAreaAsContoursFromLayer.py", line 255, in processAlgorithm
contour_featurelist = net.calcIsoContours(max_dist, interval, output_path)
File "/home/matteo/.local/share/QGIS/QGIS3/profiles/matteo/python/plugins/QNEAT3/Qneat3Framework.py", line 411, in calcIsoContours
band_in = ds_in.GetRasterBand(1)
AttributeError: 'NoneType' object has no attribute 'GetRasterBand'

Execution failed after 1.74 seconds
```

**Comment by root676** [comment]
Hi Matteo! Thanks for reporting the error!

**Comment by ghtmtt** [comment]
Hi Clemens :) Please tell me if you want a small project. I did prepare it because the error is straightforward, but just ask ;)

**Comment by root676** [comment]
:) Oh that would be very helpful! This should help identifying the faulty code sections.

**Comment by josephholler** [comment]
Line 224 of IsoAreaAsContoursFromLayer.py sets the analysis CRS using the Project CRS. This could be changed to use the network layer CRS.
In addition, it'd be nice to check consistency between the CRSs of all input layers (and the Project CRS for "from point" algorithms) and either reproject them to the network layer CRS or provide a warning.

**Comment by ghtmtt** [comment]
wow, thanks @josephholler for the fix. I'll test the related PR and let you know ASAP!

**Comment by root676** [comment]
@josephholler nice bugfix there! Thank you!

**Comment by ghtmtt** [comment]
seems fixing the issue, thanks @josephholler

**Comment by josephholler** [comment]
That's great!  Thanks to @kufreu for working on this as well.

**Comment by root676** [comment]
Implemented in 1.0.5
